tools/git-hooks/pre-commit.d/format-json.sh

```diff
@@ -4,20 +4,32 @@ set -eu
 
 cd "$(dirname "$0")/../../.."
 
-set +e
-changed_json="$(
-  git diff --cached --name-only --diff-filter=ACMR |
-  grep "\.json$"
-)"
-set -e
+echo_changed_json() {
+  set +e
+  git diff --cached --name-only --diff-filter=ACMR | grep "\.json$"
+  set -e
+}
 
-if [ -z "$changed_json" ]; then
-  exit
-fi
+format_json() {
+  echo "Formatting $(echo "$changed_json" | wc -l) json files"
+  echo "$changed_json" | while IFS= read -r file; do
+    if jq empty "$file" >/dev/null 2>&1; then
+      # shellcheck disable=SC2005
+      echo "$(jq . "$file")" >"$file"
```

Comment:
Won't this have the same problem we fixed in the es json hook? If somehow the file has invalid formatting the resulting file will be empty and the original is lost

Reply:
hence the `if` on line 16